src/Amadevus.RecordGenerator.Attributes/Features.cs

```diff
@@ -42,16 +42,44 @@ public enum Features
         /// <c>Deconstruct</c> method:
         /// <see href="https://docs.microsoft.com/pl-pl/dotnet/csharp/deconstruct#deconstructing-user-defined-types"/>
         /// </summary>
-        Deconstruct = 0b_10000,
+        Deconstruct = 0b_1000_0,
 
         /// <summary>
         /// Default feature set.
         /// </summary>
         Default = Constructor | Withers | ToString | Builder | Deconstruct,
 
-        //ObjectEquals = 0b_100000,
-        //EquatableEquals = 0b_1000000,
-        //OperatorEquals = 0b_10000000,
-        //Equality = ObjectEquals | EquatableEquals | OperatorEquals
+        /// <summary>
+        /// <see cref="object.Equals(object)"/> and <see cref="object.GetHashCode()"/> overrides. 
+        /// <see cref="object.Equals(object)"/> implements an equality comparison which compares 
+        /// all record entries properties.
+        /// of the other <see cref="object"/> with it's own values. 
+        /// <see cref="object.GetHashCode()"/> override which calculates a hash code using all record properties.
+        /// <see cref="object.GetHashCode()"/>
+        /// </summary>
+        /// <remarks>
+        /// <see cref="object.Equals(object)"/> returns <see langword="true" /> if all values match 
+        /// and returns <see langword="false"/> if <see langword="null" /> is passed, another type is passed or at least one value doesn't mach.
+        /// </remarks>
+        ObjectEquals = 0b_1000_00,
+
+        /// <summary>
+        /// <see cref="IEquatable{T}.Equals(T)"/> implementation that provides a call to the 
+        /// <see cref="object.Equals(object)"/> method in a type safe way. 
+        /// Is dependent on the feature <see cref="ObjectEquals"/>.
+        /// </summary>
+        EquatableEquals = 0b_1000_000,
+
+        /// <summary>
+        /// Overrides the <see langword="==" /> operator by using the <see cref="object.Equals(object)"/> method.
+        /// Is dependent on the feature <see cref="ObjectEquals"/>.
+        /// </summary>
+        OperatorEquals = 0b_1000_0000,
```

Comment:
I thought we agreed to use direct operator equality/valuetuple equality. That might come later.

Reply:
Moved to #75